File: Shared/Models/DocumentResult.cs
namespace Shared.Models;

public class DocumentResult
{
    public Guid Id { get; set; }
    public Guid OwnerId { get; set; }
    public string? Title { get; set; }
    public DateTime? CreatedAt { get; set; }
    public DateTime? LastModifiedAt { get; set; }
    public string? FileName { get; set; }
    public string? Result { get; set; }
    public string Discriminator { get; set; }
    
    public DocumentResult()
    {
        Discriminator = GetType().Name;
    }
}

File: DocumentDistiller/src/app/models/document-result.ts
export interface DocumentResult {
  id: string; // guid
  ownerId: string; // guid
  projectId: string; // guid
  title?: string;
  createdAt?: Date;
  lastModifiedAt?: Date;
  fileName?: string;
  result?: string;
  discriminator: string;
}

export interface UpdateDocumentResultDto {
  id: string; // guid
  title: string;
  result: string;
}
